dev: ClickHouse 21.8

lynnagara opened the pull request.

lynnagara committed:
dev: ClickHouse 21.8

MeredithAnya commented on src/sentry/conf/server.py:
```diff
@@ -2732,12 +2726,7 @@ def build_cdc_postgres_init_db_volume(settings: Any) -> dict[str, dict[str, str]
-            # Official support: https://github.com/ClickHouse/ClickHouse/issues/22222
-            # This image is build with this script https://gist.github.com/filimonov/5f9732909ff66d5d0a65b8283382590d
-            else "ghcr.io/getsentry/image-mirror-altinity-clickhouse-server:21.6.1.6734-testing-arm",
+            "image": "ghcr.io/getsentry/image-mirror-yandex-clickhouse-server:21.8.13.1",
```
Im assuming you need the altinity stable at the end https://github.com/getsentry/image-mirror/pkgs/container/image-mirror-altinity-clickhouse-server/115571514?tag=21.8.13.1.altinitystable

lynnagara committed:
fix image

lynnagara committed:
oops

asottile-sentry reviewed (approved):
noice

![](https://i.fluffy.cc/b6wlBbdVkX83CHpLWxHZw74WJxHVQc9S.gif)

lynnagara commented:
@asottile-sentry do you have any idea why all these random tests are failing?

lynnagara committed:
Merge branch 'master' into clickhouse-21.8

ryan953 committed:
feat(bug reports): Persist the chosen mailbox into the url, for easy sharing (#58804)

<img width="737" alt="inbox"
src="https://github.com/getsentry/sentry/assets/187460/4814a444-2f9f-46cd-bf48-d93a664fa8a5">

<img width="738" alt="resolved"
src="https://github.com/getsentry/sentry/assets/187460/b045df7f-f123-4138-8a91-cfcdef161eb2">
<img width="804" alt="acrhived"
src="https://github.com/getsentry/sentry/assets/187460/6a700f8d-0bab-42f3-a9ba-597bc129a1e9">

michellewzhang committed:
ref(bug reports): unread items marked by dot (#58803)

unread items are marked with a dot, which doesn't go away until the user
manually marks it as read (the dot does not automatically disappear upon
clicking the feedback detail)

<img width="411" alt="SCR-20231025-mwdi"
src="https://github.com/getsentry/sentry/assets/56095982/4b5088cc-56d2-4c23-8f55-4b3535bb4ce3">

azaslavsky committed:
feat(backup): Support decryption for `sentry compare` (#58796)

Encrypted files can now be compared as well.

Issue: getsentry/team-ospo#203

michellewzhang committed:
feat(bug reports): add seenBy list and mark read/unread functionality (#58800)

https://github.com/getsentry/sentry/assets/56095982/3169339c-0f42-43ef-8bc0-44abc969a412

JoshFerge committed:
feat(feedback): add name to metadata (#58809)

adds `name` to feedback issue metadata. Note that once ingestion is
running we'll solely rely upon the logic in event type and delete the
logic in `ingest.py` this will happen in the next few weeks.

dependabot[bot] committed:
chore(deps): bump crypto-js from 4.0.0 to 4.2.0 (#58816)

Bumps [crypto-js](https://github.com/brix/crypto-js) from 4.0.0 to
4.2.0.
<details>
<summary>Commits</summary>
<ul>
<li><a
href="https://github.com/brix/crypto-js/commit/808f499ec789fcd68416328a40b8735a5c962116"><code>808f499</code></a>
Merge branch 'release/4.2.0'</li>
<li><a
href="https://github.com/brix/crypto-js/commit/d5af3ae4130f63342e5623a22f225c612a32e34a"><code>d5af3ae</code></a>
Update release notes.</li>
<li><a
href="https://github.com/brix/crypto-js/commit/9496e07c9a727f54e98942e86dd931d27558ea02"><code>9496e07</code></a>
Bump version.</li>
<li><a
href="https://github.com/brix/crypto-js/commit/421dd538b2d34e7c24a5b72cc64dc2b9167db40a"><code>421dd53</code></a>
Change default hash algorithm and iteration's for PBKDF2 to prevent weak
secu...</li>
<li><a
href="https://github.com/brix/crypto-js/commit/d1f4f4daec4f053be3eea4f6233d1d05ccec22fa"><code>d1f4f4d</code></a>
Update grunt.</li>
<li><a
href="https://github.com/brix/crypto-js/commit/c7552897a45df2a32a595b7708addbe865a8b64d"><code>c755289</code></a>
Discontinued</li>
<li><a
href="https://github.com/brix/crypto-js/commit/1da3dabf93f0a0435c47627d6f171ad25f452012"><code>1da3dab</code></a>
Discontinued</li>
<li><a
href="https://github.com/brix/crypto-js/commit/4dcaa7afd08f48cd285463b8f9499cdb242605fa"><code>4dcaa7a</code></a>
Merge pull request <a
href="https://redirect.github.com/brix/crypto-js/issues/380">#380</a>
from Alanscut/dev</li>
<li><a
href="https://github.com/brix/crypto-js/commit/762feb23a59f5db565b1e9c7ffbe7476264b755f"><code>762feb2</code></a>
chore: rename BF to Blowfish</li>
<li><a
href="https://github.com/brix/crypto-js/commit/fb81418583004d903fbc63190be21e460a52521b"><code>fb81418</code></a>
feat: blowfish support</li>
<li>Additional commits viewable in <a
href="https://github.com/brix/crypto-js/compare/4.0.0...4.2.0">compare
view</a></li>
</ul>
</details>
<br />


[![Dependabot compatibility
score](https://dependabot-badges.githubapp.com/badges/compatibility_score?dependency-name=crypto-js&package-manager=npm_and_yarn&previous-version=4.0.0&new-version=4.2.0)](https://docs.github.com/en/github/managing-security-vulnerabilities/about-dependabot-security-updates#about-compatibility-scores)

Dependabot will resolve any conflicts with this PR as long as you don't
alter it yourself. You can also trigger a rebase manually by commenting
`@dependabot rebase`.

[//]: # (dependabot-automerge-start)
[//]: # (dependabot-automerge-end)

---

<details>
<summary>Dependabot commands and options</summary>
<br />

You can trigger Dependabot actions by commenting on this PR:
- `@dependabot rebase` will rebase this PR
- `@dependabot recreate` will recreate this PR, overwriting any edits
that have been made to it
- `@dependabot merge` will merge this PR after your CI passes on it
- `@dependabot squash and merge` will squash and merge this PR after
your CI passes on it
- `@dependabot cancel merge` will cancel a previously requested merge
and block automerging
- `@dependabot reopen` will reopen this PR if it is closed
- `@dependabot close` will close this PR and stop Dependabot recreating
it. You can achieve the same result by closing it manually
- `@dependabot show <dependency name> ignore conditions` will show all
of the ignore conditions of the specified dependency
- `@dependabot ignore this major version` will close this PR and stop
Dependabot creating any more for this major version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this minor version` will close this PR and stop
Dependabot creating any more for this minor version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this dependency` will close this PR and stop
Dependabot creating any more for this dependency (unless you reopen the
PR or upgrade to it yourself)
You can disable automated security fix PRs for this repo from the
[Security Alerts
page](https://github.com/getsentry/sentry/network/alerts).

</details>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

RyanSkonnord committed:
test(hc): Fix test_member_joined in region mode (#58811)

Run `accept_invite` in control mode and flush outbox.

john-z-yang committed:
Ref(metrics): Refactor batch to retain offset and partition of invalid messages (#58675)

Part of my efforts to add dlq to the sentry metrics consumers, see the
final change here:
https://github.com/getsentry/sentry/pull/57998 

Previous PR in the stack:
https://github.com/getsentry/sentry/pull/58474

Next PR in the stack:
https://github.com/getsentry/sentry/pull/58677

### Overview

- Refactor `batch` to maintain both offsets and partition of an invalid
message, from just maintaining the offset. This is done because we need
to produce both offset and partition to let arroyo move messages into
dlq.
- Separate messages that are invalid and filtered, in 2 separate data
structures. This is done because when we are ready to produce to dlq,
the filtered messages do not end up in the dlq.
- Implements a `_extract_message` function as help to
`_extract_messages` so the process of recovering offset can be cleaner
- Update test

evanpurkhiser committed:
fix(crons): Use localized time in missed subtitle (#58814)

Fixes https://github.com/getsentry/sentry/issues/58810

michellewzhang committed:
fix(bug reports): fix tags component (#58824)

Fix tags component rendering and add a few more tags: platform,
environment, transaction
<img width="742" alt="SCR-20231025-odfo"
src="https://github.com/getsentry/sentry/assets/56095982/84557a6c-fd36-49d6-93c5-aa023ac2cfa6">

RyanSkonnord committed:
test(hc): Fix freeze_time decorator on test classes (#58823)

Place `@freeze_time` inside the `@region_silo_test` so that it's
correctly applied on both runs.

evanpurkhiser committed:
style(crons): Add comments to Monitor fields (#58819)

malwilley committed:
fix(suspect-commits): Get selected provider from integration rather than repo (#58805)

RyanSkonnord committed:
test(hc): Mark TestDeprecationDecorator as no_silo_test (#58832)

The failure when running as control_silo_test seems to be attributable
to the `timeiter` value being reused across test cases. Because the test
uses only the DummyEndpoint and doesn't touch any persistent models, it
should be safe to mark it as no_silo_test.

lynnagara committed:
mark bad tests xfail

lynnagara commented:
I marked the bad tests XFAIL, though they should probably just be deleted

lynnagara committed:
Merge branch 'master' into clickhouse-21.8

onewland commented:
looking forward to seeing this hit 22.8 🚀
